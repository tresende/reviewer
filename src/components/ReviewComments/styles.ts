import styled, { css } from 'styled-components'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    margin-top: ${theme.spacings.xsmall};
    padding: ${theme.spacings.xsmall} 0;
  `}
`

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.medium};
    margin-top: ${theme.spacings.xsmall};
    border-radius: ${theme.border.radius};
    border: ${theme.border.default};
    border-color: ${theme.colors.gray300};
    background-color: ${theme.colors.white};
  `}
`
