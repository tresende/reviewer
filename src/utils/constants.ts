const SCORE_LABEL = ['PÃ©ssimo', 'Ruim', 'Mediano', 'Bom', 'Perfeito']
const MIN_TEXT_LENGTH = 5
const MAX_RATING = 5
const DEFAULT_SCORE = 3
const ANIMATION_DURATION = 1000

export { SCORE_LABEL, MAX_RATING, ANIMATION_DURATION, MIN_TEXT_LENGTH, DEFAULT_SCORE }
