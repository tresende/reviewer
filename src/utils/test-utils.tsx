import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
