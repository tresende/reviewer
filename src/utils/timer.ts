export function timeout(ms?: number | undefined) {
  return new Promise((resolve) => setTimeout(resolve, ms))
}
