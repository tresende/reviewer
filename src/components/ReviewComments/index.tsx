import Breadcrumb from 'components/Breadcrumb'
import CommentComponent from 'components/Comment'
import Comment from 'models/Comment'

import * as S from './styles'

export type ReviewCommentsProps = {
  comments: Comment[]
}

const ReviewComments = ({ comments }: ReviewCommentsProps) => (
  <S.Wrapper>
    <Breadcrumb />
    <S.Container>
      {comments.map((comment) => (
        <CommentComponent key={comment.id} comment={comment} />
      ))}
    </S.Container>
  </S.Wrapper>
)

export default ReviewComments
