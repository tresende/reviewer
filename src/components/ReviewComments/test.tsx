import { render, screen } from 'utils/test-utils'
import Mock from 'mocks/mock'
import ReviewComments from '.'

jest.mock('components/Comment', () => ({
  __esModule: true,
  default: function Mock() {
    return <div data-testid="comments" />
  }
}))

describe('<ReviewComments />', () => {
  it('should render component with muiltiple comments', () => {
    const { container } = render(<ReviewComments comments={Mock.comments} />)

    expect(screen.getAllByTestId('comments')).toHaveLength(5)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render component without comments', () => {
    render(<ReviewComments comments={[]} />)

    expect(screen.queryAllByTestId('comments')).toHaveLength(0)
  })
})
