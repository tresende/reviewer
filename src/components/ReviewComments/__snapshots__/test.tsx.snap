// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewComments /> should render component with muiltiple comments 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c2 {
  font-size: 1rem;
  font-weight: 700;
  font-weight: 400;
}

.c2::after {
  color: #CCCCCC;
  margin: 0 0.8rem;
  content: '>';
}

.c3 {
  font-size: 1rem;
  font-weight: 700;
}

.c0 {
  margin-top: 1.6rem;
  padding: 1.6rem 0;
}

.c4 {
  padding: 3.2rem;
  margin-top: 1.6rem;
  border-radius: 0.4rem;
  border: 1px solid;
  border-color: #D8D8D8;
  background-color: #FFFFFF;
}

<section
  class="c0"
>
  <ol
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        href="#"
      >
        BeerOrCoffee
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        href="#"
      >
        Coworkings Brasil
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        href="#"
      >
        BrasíS.Itema/DF
      </a>
    </li>
    <li
      class="c2"
    >
      Asa Sul
    </li>
    <li
      class="c2"
    >
      <a
        href="#"
      >
        BSB Coworking
      </a>
    </li>
    <li
      class="c3"
    >
      Avaliações
    </li>
  </ol>
  <div
    class="c4"
  >
    <div
      data-testid="comments"
    />
    <div
      data-testid="comments"
    />
    <div
      data-testid="comments"
    />
    <div
      data-testid="comments"
    />
    <div
      data-testid="comments"
    />
  </div>
</section>
`;
